subsys/bluetooth/host/id.c

```diff
@@ -1980,8 +1980,18 @@ int bt_id_set_adv_own_addr(struct bt_le_ext_adv *adv, uint32_t options,
 				scan_disabled = true;
 				bt_le_scan_set_enable(BT_HCI_LE_SCAN_DISABLE);
 			}
-#endif /* defined(CONFIG_BT_OBSERVER) */
+
+			/* If we are scanning with the identity address, it does
+			 * not make sense to set an NRPA.
+			 */
+			if (!(IS_ENABLED(CONFIG_BT_SCAN_WITH_IDENTITY) &&
+			    atomic_test_bit(bt_dev.flags, BT_DEV_SCANNING))) {
+				err = bt_id_set_adv_private_addr(adv);
+			}
```

Comment:
Is this the right place to have this?

Based on the naming of this function, it relates to advertisements, so why are handling a case for `CONFIG_BT_SCAN_WITH_IDENTITY` here?

Reply:
I also why `CONFIG_BT_SCAN_WITH_IDENTITY` even exists as a Kconfig instead of being a `bt_le_scan_opt` value, since it's just changing some runtime behavior anyways :o